api/account/api.go

```diff
@@ -0,0 +1,101 @@
+package account
+
+import (
+	"errors"
+	"regexp"
+
+	kc "github.com/cloudtrust/keycloak-client"
+)
+
+// UserRepresentation struct
+type AccountRepresentation struct {
+	Username    *string `json:"username,omitempty"`
+	Email       *string `json:"email,omitempty"`
+	FirstName   *string `json:"firstName,omitempty"`
+	LastName    *string `json:"lastName,omitempty"`
+	PhoneNumber *string `json:"phoneNumber,omitempty"`
+}
+
+// ConvertToAPIAccount creates an API account representation from  a KC user representation
+func ConvertToAPIAccount(userKc kc.UserRepresentation) AccountRepresentation {
+	var userRep AccountRepresentation
+
+	userRep.Username = userKc.Username
+	userRep.Email = userKc.Email
+	userRep.FirstName = userKc.FirstName
+	userRep.LastName = userKc.LastName
+
+	if userKc.Attributes != nil {
+		var m = *userKc.Attributes
+
+		if m["phoneNumber"] != nil {
+			var phoneNumber = m["phoneNumber"][0]
+			userRep.PhoneNumber = &phoneNumber
+		}
+	}
+	return userRep
+}
+
+// ConvertToKCUser creates a KC user representation from an API user
+func ConvertToKCUser(user AccountRepresentation) kc.UserRepresentation {
+	var userRep kc.UserRepresentation
+
+	userRep.Username = user.Username
+	userRep.Email = user.Email
+	userRep.FirstName = user.FirstName
+	userRep.LastName = user.LastName
+
+	var attributes = make(map[string][]string)
+
+	if user.PhoneNumber != nil {
+		attributes["phoneNumber"] = []string{*user.PhoneNumber}
+	}
+
+	if len(attributes) > 0 {
```

Comment:
Fine if you plan to add attributes... else, this can be moved in the if block above